### ISSUE TITLE
Linux: No "/etc/hosts" file in docker container

### ISSUE THREAD

**Issue by Ya3Sh** [docker/docker#12887]
I have just started with docker on RedHat Linux and when I run a "ubuntu:latest" container I don't see /etc/hosts file in the container. 
I get following error:
cat: /etc/hosts: Not a directory

That prevents me using any application that is trying to get hostname programmatically 

Docker version I am using is:
Client version: 1.6.0
Client API version: 1.18
Go version (client): go1.4.2
Git commit (client): 4749651
OS/Arch (client): linux/amd64
Server version: 1.6.0
Server API version: 1.18
Go version (server): go1.4.2
Git commit (server): 4749651
OS/Arch (server): linux/amd64

**Comment by LK4D4** [comment]
Probably selinux stuff.

**Comment by yateenshaha** [comment]
I have it disabled.

**Comment by jfrazelle** [comment]
can you see if its the same problem as https://github.com/docker/docker/issues/12613

**Comment by jfrazelle** [comment]
also we need all the information gordon requested

**Comment by Ya3Sh** [comment]
@jfrazelle The problem #12613 doesn't look similar though its also a networking issue. I never had it run successfully on RHEL even on docker version 1.5.0. Then I tried manually upgrading to dokcer version 1.6.0 with no luck.
Here is the information @GordonTheTurtle requested.

Description of problem:
Bashing into a docker container say "ubuntu:latest" and executing /etc/hosts gives following error:
cat: /etc/hosts: Not a directory

`docker version`:
Client version: 1.6.0
Client API version: 1.18
Go version (client): go1.4.2
Git commit (client): 4749651
OS/Arch (client): linux/amd64
Server version: 1.6.0
Server API version: 1.18
Go version (server): go1.4.2
Git commit (server): 4749651
OS/Arch (server): linux/amd64

`docker info`: sudo docker info
Storage Driver: devicemapper
 Pool Name: docker-0:21-8690-pool
 Pool Blocksize: 65.54 kB
 Backing Filesystem: nfs
 Data file:
 Metadata file:
 Data Space Used: 8.682 GB
 Data Space Total: 107.4 GB
 Data Space Available: 98.69 GB
 Metadata Space Used: 14.79 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.133 GB
 Udev Sync Supported: false
 Library Version: 1.02.82-git (2013-10-04)
Execution Driver: native-0.2
Kernel Version: 2.6.32-504.16.2.el6.x86_64
Operating System: <unknown>
CPUs: 2
Total Memory: 15.58 GiB



`uname -a`:
Linux hostname 2.6.32-504.16.2.el6.x86_64 x86_64 x86_64 x86_64 GNU/Linux

Environment details (AWS, VirtualBox, physical, etc.):
Physical

How reproducible:
sudo docker run -it ubuntu:latest bash
# cat /etc/hosts

Steps to Reproduce:
1.
2.
3.


Actual Results:
cat: /etc/hosts: Not a directory

Expected Results:
Should have apt hosts file to support networking.

**Comment by visualphoenix** [comment]
Try with the following rpm: https://kojipkgs.fedoraproject.org/packages/docker-io/1.6.0/1.el6/x86_64/docker-io-1.6.0-1.el6.x86_64.rpm

Your problem was not reproducible for me using this rpm (for centos/rhel 6.6+ hosts).

**Comment by Ya3Sh** [comment]
Thanks @visualphoenix. It works for me.
The issue appears to be with changing docker working directory. I continue to see this issue if I change docker working directory to a *mounted* file system. It works if I change working directory to physical disk. Is it not supported?

**Comment by corvis** [comment]
Hi @Ya3Sh,
Looks like I have exactly the same issue, did you find any solution which allows to keep ```-g``` to point mounted file system?

**Comment by Ya3Sh** [comment]
@corvis,
No. It seems not supported though I have not heard a word about it from docker. I had to move on.

**Comment by visualphoenix** [comment]
@Ya3Sh @corvis might be worth looking into this again once 1.7.1/1.8.0 comes out with 6.6+ fixes and the new docker volume backend. I'm not sure what issues might come up relocating the graph as you are - there might be some subtle interplay between docker and how dm thin provisioning is working with what you're trying to do... Maybe you could be more specific as to the goal you were trying to achieve with relocating the graph? You said:

```
other_args="-g /data/dockerws"
```

What does the mount info for that look like?

**Comment by cpuguy83** [comment]
Are you moving the docker data to the new location?

**Comment by gf-atebbe** [comment]
I'm running into this issue as well.  I moved the docker data to the new location before starting docker.  My data directory is an NFS mount because the local disk is too small.  

other_args="-g /mnt/docker/service/popper-3/"

```
[root@popper-3 ~]# docker run --rm -ti debian bash
root@8dbc499d189c:/# cat /etc/hosts
cat: /etc/hosts: Not a directory
root@8dbc499d189c:/# cat /etc/resolv.conf 
cat: /etc/resolv.conf: Not a directory
```

docker version:
```
Client version: 1.7.1
Client API version: 1.19
Go version (client): go1.4.2
Git commit (client): 786b29d
OS/Arch (client): linux/amd64
Server version: 1.7.1
Server API version: 1.19
Go version (server): go1.4.2
Git commit (server): 786b29d
OS/Arch (server): linux/amd64
```

docker info:
```
Containers: 4
Images: 119
Storage Driver: devicemapper
 Pool Name: docker-0:44-2143148-pool
 Pool Blocksize: 65.54 kB
 Backing Filesystem: nfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 6.186 GB
 Data Space Total: 107.4 GB
 Data Space Available: 101.2 GB
 Metadata Space Used: 7.619 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.14 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Data loop file: /mnt/docker/service/popper-3/devicemapper/devicemapper/data
 Metadata loop file: /mnt/docker/service/popper-3/devicemapper/devicemapper/metadata
 Library Version: 1.02.89-RHEL6 (2014-09-01)
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 2.6.32-504.1.3.el6.x86_64
Operating System: <unknown>
CPUs: 32
Total Memory: 126 GiB
Name: popper-3
ID: UVSS:IKOY:6ZEW:D5IN:INEJ:TKFR:FP7U:RPHU:YGW3:GGNL:ZX7K:NWU5
```

uname -a:
```
Linux popper-3 2.6.32-504.1.3.el6.x86_64 #1 SMP Tue Nov 11 14:19:04 CST 2014 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by justincormack** [comment]
You cannot run devicemapper on nfs directly as far as I know.
